.github/workflows/update-types.yml

```diff
@@ -24,82 +28,110 @@ jobs:
-        name: Run generator
-        run: |
-          if [ -z "${{ github.event.inputs.single_path }}" ]
-          then
-            npm run generate -- --specs-dir ../../workflow-temp/azure-rest-api-specs
-          else
-            npm run generate -- --specs-dir ../../workflow-temp/azure-rest-api-specs --single-path ${{ github.event.inputs.single_path }}
-          fi
-
-          summary="$(<../../generated/summary.log)"
-          summary="${summary//'%'/'%25'}"
-          summary="${summary//$'\n'/'%0A'}"
-          summary="${summary//$'\r'/'%0D'}"
-
-          echo "::set-output name=summary::$summary"
-        working-directory: src/generator
-
-      - id: get_swagger_gh_uri
-        name: Get GitHub URI for azure-rest-api-specs
-        run: |
-          git_sha=`git rev-parse HEAD`
-          echo "::set-output name=gh_uri::https://github.com/Azure/azure-rest-api-specs/tree/$git_sha"
-        working-directory: workflow-temp/azure-rest-api-specs
-
-      - name: Create Pull Request
-        uses: peter-evans/create-pull-request@v5
+      - name: Tar+Gzip all json files under the generated folder
+        run: find . -name '*.json' -print0 | tar -czf ../types.tgz --null --files-from -
+        working-directory: generated
+
+      - name: List all files in the generated directory
+        run: ls -la
+
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+
+      - name: Publish OCI Artifact
+        uses: oras-project/setup-oras@v1
         with:
-          committer: GitHub <noreply@github.com>
-          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
-          signoff: false
-          branch: autogenerate
-          branch-suffix: short-commit-hash
-          delete-branch: true
-          title: |
-            Update Generated Types ${{ github.event.inputs.single_path && format('(single path: {0})', github.event.inputs.single_path) || '' }}
-          body:  |
-            Update Generated Types ${{ github.event.inputs.single_path && format('(single path: {0})', github.event.inputs.single_path) || '' }}
-
-            Generate types for ${{ steps.get_swagger_gh_uri.outputs.gh_uri }}
-
-            Summary
-            ${{ steps.generate.outputs.summary }}
-          commit-message: |
-            Update Generated Types ${{ github.event.inputs.single_path && format('(single path: {0})', github.event.inputs.single_path) || '' }}
-
-            Generate types for ${{ steps.get_swagger_gh_uri.outputs.gh_uri }}
-          labels: autogenerate
-          draft: false
+          version: 1.0.0
+      - run: >-
+          oras push {{ secrets.PUBLISH_REGISTRY_SERVER }}/bicep/providers/az:protocolv1,${{ steps.date.outputs.date }} 
+          --annotation bicep.serialization.format=v1 
+          --artifact-type application/vnd.ms.bicep.provider.types.v1+json 
+          generated\types.tgz:application/vnd.ms.bicep.provider.types.v1.tar+gzip
```

Comment:
```suggestion
          generated/types.tgz:application/vnd.ms.bicep.provider.types.v1.tar+gzip
```

Reply:
A few thoughts:
* Do we need a version on the artifact type? For modules, we currently use `application/vnd.ms.bicep.module.artifact`. We can still have versioning in layers, but wondering if we ever need to distinguish between artifact types of different versions before the manifest is pulled (for example, when querying for OCI repository names via the ACR metadata search service).
* For consistency with module artifact type, we should call the artifact type for provider either `application/vnd.ms.bicep.provider.artifact` or `application/vnd.ms.bicep.provider.artifact.v1`, depending on the answer to the previous question. I don't think we need the `types` part because that can be done via layers. Our provider artifacts in the future may end up delivering executables in addition to types as well. (possible direction for 3rd party extensibility)
* For consistency with module media types, we should call the layer media type `application/vnd.ms.bicep.provider.types.layer.v1+json`. We may have provider-related non-types layers in a provider artifact in the future.

Reply:
> * Do we need a version on the artifact type? For modules, we currently use `application/vnd.ms.bicep.module.artifact`. We can still have versioning in layers, but wondering if we ever need to distinguish between artifact types of different versions before the manifest is pulled (for example, when querying for OCI repository names via the ACR metadata search service).

I think we need a version for the artifact type because over time we may want to evolve the contents to be serialized in a different way and perhaps change the structure of how the types are packaged. That being said if you feel strongly about this I am happy to remove the version.

> * For consistency with module artifact type, we should call the artifact type for provider either application/vnd.ms.bicep.provider.artifact or application/vnd.ms.bicep.provider.artifact.v1, depending on the answer to the previous question. I don't think we need the types part because that can be done via layers. Our provider artifacts in the future may end up delivering executables in addition to types as well. (possible direction for 3rd party extensibility)

I opted for option 1, happy to change this to option 2 if it feels like a better option. I am not sure what are the drawbacks/benefits of one vs the other so I would appreciate more clarity on that.

> * For consistency with module media types, we should call the layer media type application/vnd.ms.bicep.provider.types.layer.v1+json. We may have provider-related non-types layers in a provider artifact in the future.

I adopted the naming convention that you recommended, I kept it application/vnd.ms.bicep.provider.types.layer.v1.tar+gzip because the artifact is a gzipped tar. I don't think `json` is matching but I am willing to maybe use zip instead of tar+gzip and have the contents be a zip file if you think that makes it more compliant, let me know